src/main/java/org/truffleruby/language/arguments/RubyArguments.java

```diff
@@ -115,31 +118,33 @@
         return newArgs;
     }
 
-    public static boolean assertFrameArguments(Object[] frameArguments) {
-        Object callerData = getCallerData(frameArguments);
-        InternalMethod method = getMethod(frameArguments);
-        DeclarationContext declarationContext = getDeclarationContext(frameArguments);
-        Object self = getSelf(frameArguments);
-        Object block = getBlock(frameArguments);
-        Object[] arguments = getArguments(frameArguments);
-
-        assert method != null;
-        assert declarationContext != null;
-        assert self != null;
+    public static boolean assertArguments(Object[] arguments) {
         assert arguments != null;
-        assert ArrayUtils.assertValidElements(arguments, arguments.length);
-
-        assert callerData == null ||
-                callerData instanceof MaterializedFrame ||
-                callerData instanceof SpecialVariableStorage ||
-                callerData instanceof FrameAndVariables;
-
-        /* The block in the frame arguments is always either a Nil or RubyProc. The provision of Nil if the caller
-         * doesn't want to provide a block is done at the caller, because it will know the type of values within its
-         * compilation unit.
-         *
-         * When you read the block back out in the callee, you'll therefore get a Nil or RubyProc. */
-        assert block instanceof Nil || block instanceof RubyProc : block;
+        assert arguments.length >= RUNTIME_ARGUMENT_COUNT;
+
+        final Object declarationFrame = arguments[ArgumentIndicies.DECLARATION_FRAME.ordinal()];
```

Comment:
Why not using the various helper methods here like `getDeclarationFrame()`?

Reply:
There's no such method that takes `Object[]` - I'd have to write one.

Reply:
That's completely fine, there is already one for basically every other internal argument.

Reply:
I'm ending up writing weird code like.

```
getDeclarationFrame(arguments); // value not used - just calling for the side effect of the possibly failing cast
```

Reply:
`final Object declarationFrame = getDeclarationFrame(arguments);` no? Just like the code already did before.

Reply:
Ah, I get it, the assert becomes completely redundant in that case.

Reply:
OK, fine to keep as-is, I didn't understand the motivation but now I do.